src/clj_kondo/impl/analysis/java.clj

```diff
@@ -64,18 +79,52 @@
      ClassReader/SKIP_DEBUG)
     @result*))
 
-(defn source->class-name [file]
-  (let [fname (entry->class-name file)
-        fname (str/replace fname "\\" ".")
-        class-name (last (str/split fname #"\."))]
-    (with-open [file-reader (io/reader file)]
-      (binding [*in* file-reader]
-        (loop []
-          (if-let [next-line (read-line)]
-            (if-let [[_ package] (re-matches #"\s*package\s+(\S*)\s*;\s*" next-line)]
-              (str package "." class-name)
-              (recur))
-            class-name))))))
+(defn ^:private node->flag-member-type [node]
+  (condp = (type node)
+    FieldDeclaration :field
+    MethodDeclaration :method
+    ConstructorDeclaration :method))
```

Comment:
Why is this `:method` and not `:constructor`?

Reply:
I followed the same rule we had at ASM, call it method and know it by the name, should we change it?

Reply:
From ASM eyes, a constructor is a method as well, we could infer that and make javaparser returns constructor as well, but maybe do in another PR, the way it's is consistent always returning method for both asm and javaparser